sqlmesh/utils/transactional_file.py

```diff
@@ -136,7 +128,9 @@ def release_lock(self) -> None:
             self._original_contents = None
 
     @contextmanager
-    def read_with_lock(self, timeout: float = 61.0) -> t.Iterator[t.Optional[bytes]]:
+    def read_with_lock(
```

Comment:
I don't like this API. I think the context manager API should be `def with_lock()` to safely acquire and release the lock, and then we can do reading / writing / whatever else inside.